public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -166,13 +169,41 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     this.setState({ isEditing: true });
 
     // Propagate change edit mode change to children
+    this.propagateEditModeChange();
+    this.startTrackingChanges();
+  };
+
+  public saveCompleted(saveModel: Dashboard, result: SaveDashboardResponseDTO, folderUid?: string) {
+    this._initialSaveModel = {
+      ...saveModel,
+      id: result.id,
+      uid: result.uid,
+      version: result.version,
+    };
+
+    this.stopTrackingChanges();
+    this.setState({
+      version: result.version,
+      isDirty: false,
+      uid: result.uid,
+      id: result.id,
+      meta: {
+        ...this.state.meta,
+        uid: result.uid,
+        url: result.url,
+        slug: result.slug,
+        folderUid: folderUid,
+      },
+    });
+    this.startTrackingChanges();
```

Comment:
I think we should also exit Edit mode. It is a bit confusing that after saving the Save button is still available.

Reply:
@dprokop I had that before but don't think that is a good idea. Think you should stay in edit mode. You might have more edits to do just because you saved does not mean you are done.